src/DDTrace/Span.php

```diff
@@ -392,26 +391,6 @@ public function getAllBaggageItems()
         return $this->context->baggageItems;
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @param Integration $integration
-     * @return self
-     */
-    public function setIntegration(Integration $integration)
-    {
-        $this->integration = $integration;
-        return $this;
-    }
-
-    /**
-     * @return null|Integration
-     */
-    public function getIntegration()
-    {
-        return $this->integration;
-    }
```

Comment:
Are these changes related to removing spl_autoload_register and changing composer or just something you did while passing by?

Reply:
they are related to it.

1. In order to remove `spl_autoload_register` we need to deliver the 'public'/noop api via composer.
2. As part of it interfaces in `DDTrace\Contracts` have to be moved to the public api
3. Span and SpanContext interface where depending on the concrete `Integration` class, which has a lot of things and we are about to remove with the work on simplifying the integrations.
4. I did not want to 'leak' that concrete class into the public API.